apps/gateway/src/middlewares/models/gemini.ts

```diff
@@ -1,14 +1,96 @@
+import { BadRequestError } from "@hebo/shared-api/errors";
 
+import type { OpenAICompatibleReasoning } from "~gateway/utils/openai-compatible-api-schemas";
 
 import { ModelAdapterBase } from "./model";
 
+import type { ProviderOptions } from "@ai-sdk/provider-utils";
 
 export abstract class GeminiModelAdapter extends ModelAdapterBase {
   readonly modality = "chat";
   readonly owned_by = "google";
   readonly pricing = {
     monthly_free_tokens: 0,
   };
+
+  transformOptions(options?: ProviderOptions): ProviderOptions {
+    const modelConfig: Record<string, any> = {};
+
+    if (options?.openaiCompatible) {
+      const reasoning = (options.openaiCompatible as any)
```

Comment:
The classes need to know external conventions like `openAiCompatible` and `modelConfig` which leads to risk in implementation mistakes. Instead, inject just the sub-object into transformOptions and handle naming conventions / object hierachy at the caller side. Also reduces boilerplate in the implementation code of transformOptions hooks.

(Same applies to transformOptions in the providers)